crates/refunder/src/refund_service.rs

```diff
@@ -136,25 +143,42 @@ impl RefundService {
                             return None;
                         }
                     };
-                    let is_refunded: bool = order_owner == Some(INVALIDATED_OWNER);
-                    Some((eth_order_placement.uid, is_refunded))
+                    let refund_status = match order_owner {
+                        Some(bytes) if bytes == INVALIDATED_OWNER => RefundStatus::Refunded,
+                        Some(bytes) if bytes == H160::zero() => RefundStatus::Invalid,
+                        // any other owner
+                        _ => RefundStatus::NotYetRefunded,
+                    };
+                    Some((eth_order_placement.uid, refund_status))
                 }
             })
             .collect::<Vec<_>>();
 
         batch.execute_all(MAX_BATCH_SIZE).await;
         let uid_with_latest_refundablility = futures::future::join_all(futures).await;
-        type TupleWithRefundStatus = (Vec<(OrderUid, bool)>, Vec<(OrderUid, bool)>);
-        let (refunded_uids, to_be_refunded_uids): TupleWithRefundStatus =
+        type TupleWithRefundStatus = (Vec<(OrderUid, RefundStatus)>, Vec<(OrderUid, RefundStatus)>);
+        let (refunded_uids, uid_with_latest_refundablility): TupleWithRefundStatus =
             uid_with_latest_refundablility
                 .into_iter()
                 .flatten()
-                .partition(|(_, is_refunded)| *is_refunded);
+                .partition(|(_, refund_status)| *refund_status == RefundStatus::Refunded);
+        let (to_be_refunded_uids, invalid_uids): TupleWithRefundStatus =
+            uid_with_latest_refundablility
+                .into_iter()
+                .partition(|(_, refund_status)| *refund_status == RefundStatus::NotYetRefunded);
         let refunded_uids: Vec<OrderUid> = refunded_uids.into_iter().map(|(uid, _)| uid).collect();
         let to_be_refunded_uids: Vec<OrderUid> = to_be_refunded_uids
             .into_iter()
             .map(|(uid, _)| uid)
             .collect();
+        if !invalid_uids.is_empty() {
+            // In exceptional cases, e.g. if the refunder tries to refund orders from a previous contract,
+            // the order_owners could be zero
+            tracing::warn!(
+                "Following ethflow uids have a invalid owner (=0x0..0) stored onchain: {:?}",
```

Comment:
Same nit: it sounds like having a zero owner is in general unexpected behavior, but this is fine and the real issue is that we try to invalidate some with zero owners. My suggestion:

```suggestion
                "Trying to invalidate orders that weren't created in the current contract. Uids: {:?}",
```